client/blocks/followers-count/index.jsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External Dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Count from 'components/count';
+import { getSelectedSite } from 'state/ui/selectors';
+
+class FollowersCount extends Component {
+	render() {
+		const { site, translate } = this.props;
+
+		if ( ! site || ! site.subscribers_count ) {
+			return null;
```

Comment:
There are two different arguments here:

The need for the component to be connected – the main reason for that was making it super-simple to use and drop in another component. I don’t think we should affect the _interface_ simplicity for the ease of writing tests.

The component accepting the whole `site` object – you’re totally on point here. Both for reducing the footprint of the data we send and, more importantly, for making sure that we don’t re-render the component all the time. `connect` returns a pure component, which prevents unnecessary re-renders using only a shallow comparison of props. In our case the `site` prop would always be a different object, because JavaScript [compares objects by reference](http://adripofjavascript.com/blog/drips/object-equality-in-javascript.html).